src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessor.cs

```diff
@@ -36,6 +36,9 @@ public async Task RunAsync(EventProcessorClient processor, CancellationToken ct)
         {
             EnsureArg.IsNotNull(processor);
 
+            // Reset previous checkpoints corresponding to an older source event hub (i.e. applicable if the source event hub changes)
+            await _checkpointClient.ResetCheckpointsAsync();
```

Comment:
@wi-y  @dustinburson Since we are dealing with Async methods, what do you think about starting to use CancellationTokens where needed? For instance, the `CancellationToken` in scope here could be passed into `ResetCheckpointsAsync` and used in the subsequent async calls.

Reply:
We should update and add proper cancellation token support but  lets create a backlog item and do it outside this PR.

Reply:
Agreed, I thought of using it for this call, but given the pattern in this file didn't seem to be doing that, decided to defer for now.
Created this WI -https://microsofthealth.visualstudio.com/Health/_workitems/edit/83006

Reply:
Updated WI - https://microsofthealth.visualstudio.com/Health/_workitems/edit/83122